domain-model-assistant/Assets/Components/Scripts/Diagram.cs

```diff
@@ -11,419 +11,549 @@
+    public void LoadJson(string cdmJson)
     {
-      if (idsToClassesAndLayouts.ContainsKey(contVal.key))
-      {
-        idsToClassesAndLayouts[contVal.key][1] = contVal;
-      }
-    });
+        ResetDiagram();
+        var classDiagram = JsonUtility.FromJson<ClassDiagramDTO>(cdmJson);
 
-    _namesToRects.Clear();
+        //store attributes to class in a dictionary
+        classDiagram.classes.ForEach(cls => this.idsToClassesAndAttributes[cls._id] = cls.attributes);
+        //store attribute types. Map type id to eclass tye
+        classDiagram.types.ForEach(type =>
+        {
+            //get appropriate substring to match
+            string res = type.eClass.Substring(type.eClass.LastIndexOf('/') + 1);
+            if (!idsToTypes.ContainsKey(type._id))
+            {
+                this.idsToTypes[type._id] = res;
+            }
+        });
+        // maps each _id to its (class object, position) pair 
+        var idsToClassesAndLayouts = new Dictionary<string, List<object>>();
+
+        classDiagram.classes.ForEach(cls => idsToClassesAndLayouts[cls._id] = new List<object> { cls, null });
+        classDiagram.layout.containers[0].value/*s*/.ForEach(contVal =>
+        {
+            if (idsToClassesAndLayouts.ContainsKey(contVal.key))
+            {
+                idsToClassesAndLayouts[contVal.key][1] = contVal;
+            }
+        });
+        _namesToRects.Clear();
+
+        foreach (var keyValuePair in idsToClassesAndLayouts)
+        {
+            var _id = keyValuePair.Key;
+            var clsAndContval = keyValuePair.Value;
+            var cls = (Class)clsAndContval[0];
+            var layoutElement = ((ElementMap)clsAndContval[1]).value;
+            _namesToRects[cls.name] = CreateCompartmentedRectangle(
+                _id, cls.name, new Vector2(layoutElement.x, layoutElement.y));
+
+        }
+        _namesUpToDate = false;
+    }
 
-    foreach (var keyValuePair in idsToClassesAndLayouts)
+    public void AddAttributes(GameObject sect, int i)
     {
-      var _id = keyValuePair.Key;
-      var clsAndContval = keyValuePair.Value;
-      var cls = (Class)clsAndContval[0];
-      Debug.Log(clsAndContval[1]);
-      var layoutElement = ((ElementMap)clsAndContval[1]).value;
-      _namesToRects[cls.name] = CreateCompartmentedRectangle(
-          _id, cls.name, new Vector2(layoutElement.x, layoutElement.y));
+        int first = 0;
+        if (i == first)
+        {
+            var compId = sect.GetComponent<Section>().GetCompartmentedRectangle()
+                            .GetComponent<CompartmentedRectangle>().ID;
+            foreach (var attr in idsToClassesAndAttributes[compId])
+            {
+                string type = GetAttributeType(idsToTypes[attr.type]);
+                sect.GetComponent<Section>().AddAttribute(attr._id, attr.name, type);
+            }
+        }
     }
 
-    _namesUpToDate = false;
-  }
+    public string GetAttributeType(string substring)
+    {
```

Comment:
This can be implemented in one line!

```cs
return attributeName.Replace("CD", "").ToLower();
```

You can cache these results to avoid doing the string replacement each time.